kotlin-analysis-api/src/main/kotlin/com/google/devtools/ksp/impl/ResolverAAImpl.kt

```diff
@@ -318,8 +320,8 @@ class ResolverAAImpl(
         val kotlinClass = classBinaryCache.getKotlinBinaryClass(virtualFile) ?: return container.declarations
         val declarationOrdering = DeclarationOrdering(kotlinClass)
 
-        return (container.declarations as? Sequence<AbstractKSDeclarationImpl>)
-            ?.sortedWith(declarationOrdering.comparator) ?: container.declarations
+        return container.declarations
+            .sortedWith(compareBy(declarationOrdering.comparator) { it as AbstractKSDeclarationImpl })
```

Comment:
explanation:
This was an unchecked cast (generics involved) and would always succeed - so replacing `as?` with `as` shouldn't change any behavior.